### PR TITLE
Feat button

### PR DESCRIPTION
**Pull request by Bhakti199**
added different types of buttons
1. primary button
2. secondary button
3. icon button
4. link button
5. button with icons
6. flaoting action button

### CODE REVIEW

File: button/button.css [hunk 1]
```diff
@@ -0,0 +1,191 @@
+.button-intro {
+  font-size: 1.25rem;
+  padding: 1rem 0 1rem 0;
```
Comment (supminn) [inline button/button.css]: `padding: 1rem 0;`

File: button/button.css [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+/* solid buttons */
+
+.ranger-primary-button-sm,
+.ranger-primary-button-md,
+.ranger-primary-button-lg,
+.ranger-btn-with-icon-button-sm,
+.ranger-btn-with-icon-button-md,
+.ranger-btn-with-icon-button-lg {
```
Comment (supminn) [inline button/button.css]: either use `btn` or `button`

File: button/button.css [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  border: none;
+  color: var(--white1);
```
Comment (supminn) [inline button/button.css]: prefer using `off-white` instead of numbering

File: button/button.css [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  flex-direction: column;
+  align-items: center;
+  gap: 2rem;
+  height: 16rem;
+  margin-bottom: 1rem;
+}
+
+/* solid buttons */
+
+.ranger-primary-button-sm,
```
Comment (supminn) [inline button/button.css]: omit using library name (ranger) across all the class names

File: button/button.css [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+/* Outlined buttons */
+.range-secondary-button-sm,
+.range-secondary-button-md,
+.range-secondary-button-lg {
+  border: 2px solid;
+  background-color: transparent;
+  font-size: 1.25rem;
+}
+.range-secondary-button-sm {
+  border-color: var(--greyFour);
```
Comment (supminn) [inline button/button.css]: Maintain a single way of naming colors. Prefer to use shade names rather than numbers.

File: button/button.css [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+  color: var(--black);
+  width: 8rem;
+  padding: 1rem 2rem 1rem 2rem;
```
Comment (supminn) [inline button/button.css:75]: `padding: 1rem 2rem`

File: button/button.css [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+}
+.button-iframes-button-with-icon {
+  width: 37rem;
+  height: 20rem;
+  margin-bottom: 1rem;
+}
+.floating-button-iframe {
+  width: 37rem;
+  height: 7rem;
+}
```
Comment (supminn) [inline button/button.css]: minimize using repeated properties. You could combine width and margin-bottom for these classes.

File: button/button.css [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+.fa-arrow-left,
+.fa-arrow-right {
+  font-size: 1rem;
+  color: var(--white1);
+}
+.fa-times-circle,
+.fa-shopping-cart,
+.fa-bolt {
+  padding-right: 5px;
```
Comment (supminn) [inline button/button.css:163]: use margin spacing instead of padding.

File: button/button.html [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+                Cards
+            </a>
+        </li>
+        <li class="sidebar-list-items">
+            <a href="" class="sidebar-items">
+                Image
+            </a>
+        </li>
+        <li class="sidebar-list-items">
+            <a href="" class="sidebar-items">
```
Comment (supminn) [inline button/button.html]: `<a href="#" class="sidebar-items"> placeholder`

File: button/button.html [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            </a>
+        </li>
+    </ul>
+
+    <div class="main-content">
+        <h1 class="main-heading">Button</h1>
+        <hr class="line-separate">
+        <p class="button-intro">Buttons means an operation which are used to trigger the corresponding logic.</p>
+       
+        <!-- solid buttons -->
```
Comment (supminn) [inline button/button.html]: you could omit these comments. They do not add any value.

File: button/button.html [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+            <button class="range-secondary-button-lg">Buy now</button>
+        </div>
+        <iframe src="https://www.thiscodeworks.com/embed/61fa771bb783be0015bbae9d" class="button-iframes" frameborder="0"></iframe>
+        <hr class="line-separate">
+
+        <!-- icons button -->
+        <h2>Icon buttons</h2>
+            <div class="icon-button-container">
+                <button class="icon-button backward-button"><i class="fas fa-arrow-left"></i></button>
+                <button class="icon-button forward-button"><i class="fas fa-arrow-right"></i></button>
```
Comment (supminn) [inline button/button.html]: `backward-button` and `forward-button` should be removed.

File: button/button.html [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+        <iframe src="https://www.thiscodeworks.com/embed/61fa771bb783be0015bbae9d" class="button-iframes" frameborder="0"></iframe>
+        <hr class="line-separate">
+
+        <!-- icons button -->
+        <h2>Icon buttons</h2>
+            <div class="icon-button-container">
+                <button class="icon-button backward-button"><i class="fas fa-arrow-left"></i></button>
+                <button class="icon-button forward-button"><i class="fas fa-arrow-right"></i></button> 
+            </div>
+        <iframe src="https://www.thiscodeworks.com/embed/61fa8445b783be0015bbaeae" class="button-iframes-icon-button" frameborder="0"></iframe>
```
Comment (supminn) [inline button/button.html]: `button-iframes-icon-button` 
the word `button` is repeated.

File: documentation.html [hunk 13]
```diff
@@ -65,7 +65,7 @@
         <a href="" class="sidebar-items">Badge</a>
       </li>
       <li class="sidebar-list-items">
-        <a href="" class="sidebar-items">Button</a>
+        <a href="/button/button.html" class="sidebar-items">Button</a>
       </li>
       <li class="sidebar-list-items">
         <a href="" class="sidebar-items">Badge</a>
```
Comment (supminn) [inline documentation.html]: class names could be named better.
`sidebar-list-item` and `sidebar-link` could be used instead.

File: style.css [hunk 14]
```diff
@@ -1,5 +1,6 @@
 @import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@500;600&display=swap");
 @import url("https://fonts.googleapis.com/css2?family=Great+Vibes&display=swap");
+@import url(button/button.css);
```
Comment (supminn) [inline style.css]: Import `style.css` within button.css

File: style.css [hunk 15]
```diff
@@ -9,14 +10,19 @@ body {
...
   --teal3: #f0fdfa;
+  --red1: #dc2626;
   --alert-green: #d9f99d;
   --alert-blue: #bfdbfe;
   --alert-yellow: #fde68a;
   --alert-red: #fecaca;
   --white1: #fafaf9;
+  --yellow2: #fbbf24;
+  --orange2: #fb923c;
+  --blue2: #60a5fa;
```
Comment (supminn) [inline style.css]: 1. Naming conventions for the colors -- based on their shade.
2. move them :root instead of `body`

**Review by supminn — CHANGES_REQUESTED** [review]
I liked your approach of building the overall structure of your library. Nice to see that you have maintained constant `rem` units.
You could improve upon the naming conventions of classes.

File: alert/alert.html [hunk 16]
```diff
@@ -18,55 +18,55 @@
...
+            <a href="/button/button.html" class="sidebar-link">
                 Button
             </a>
         </li>
-        <li class="sidebar-list-items">
-            <a href="" class="sidebar-items">
-                Card
+        <li class="sidebar-list-item">
+            <a href="" class="sidebar-link">
+                Cards
```
Comment (tusharkandpal) [inline alert/alert.html]: can go with singular here too

File: button/button.css [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+.primary-btn-md,
+.primary-btn-lg,
+.btn-with-icon-sm,
+.btn-with-icon-md,
+.btn-with-icon-lg {
+  border: none;
+  color: var(--dark-black);
+  border: none;
+  font-size: 1.25rem;
+}
```
Comment (tusharkandpal) [inline button/button.css:28]: can define a separate class for this styling

File: button/button.html [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+            </a>
+        </li>
+        <li class="sidebar-list-item">
+            <a href="/button/button.html" class="sidebar-link">
+                Button
+            </a>
+        </li>
+        <li class="sidebar-list-item">
+            <a href="" class="sidebar-link">
+                Cards
```
Comment (tusharkandpal) [inline button/button.html]: same as above

**Review by tusharkandpal — CHANGES_REQUESTED** [review]
suggested minor changes. Well structured & clean code.

File: button/button.css (hunk 17 above)
Comment (Bhakti199) [reply button/button.css:28]: I got reviewed this, its good practice.

File: button/button.html (hunk 18 above)
Comment (Bhakti199) [reply button/button.html]: I didn't get you here. Can you explain

File: button/button.css (hunk 17 above)
Comment (tusharkandpal) [reply button/button.css:28]: alright

File: button/button.html (hunk 18 above)
Comment (tusharkandpal) [reply button/button.html]: use Card here..as other components are mentioned i.e singular

Comment (Bhakti199) [reply button/button.html]: ok got it.

File: style.css (hunk 15 above)
Comment (Bhakti199) [reply style.css]: I did all the necessary changes here @supminn , can you please check it out.

**Review by tusharkandpal — APPROVED** [review]

File: style.css (hunk 15 above)
Comment (supminn) [reply style.css]: The changes look fine to me.

**Pull request merged**